launch.go

```diff
@@ -268,6 +268,11 @@ func createEnvironment(base map[string]string, secrets screwdriver.Secrets) []st
 	for k, v := range combined {
 		envStrings = append(envStrings, strings.Join([]string{k, v}, "="))
 	}
+
+	for k, v := range build.Environment {
```

Comment:
We should probably have some discussion about what env variables we do, and what we don't want the user to have influence over. This would allow the user to override all the default env settings in their sd.yaml if they wanted to.